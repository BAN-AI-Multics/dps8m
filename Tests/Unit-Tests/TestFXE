de DSBR.ADDR 01000
de dsbr.u 1
;de dsbr.addr 020000
de ppr.ic 0
de ppr.prr 0
de ppr.psr 0
de ppr.p 1
dpsinit segment table
dpsinit SDWAM

;load Play/Tests/PlayA.oct as segment 2 deferred
;load Play/Tests/UtilsA.oct as segment 3 deferred
load mainA.oct segment 0 address 0
load PlayA.oct deferred
load UtilsA.oct deferred
load simh_operators.oct deferred
load power_of_two.oct deferred
segments lot create
segments create stack 0
segments resolve
segments load deferred
segments lot snap
;q
;load Play/Tests/TestEIS.oct
;set debug stderr
;set cpu debug=trace;regdumpaqi;regdumpidx;addrmod;appending;REGDUMPPR
g 0
q

