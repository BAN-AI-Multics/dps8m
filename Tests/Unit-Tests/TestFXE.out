
dps-8/m simulator V4.0-0 Beta        git commit id: f386776a
zero-initialized segments 0 .. 7
zero-initialized SDWAM
segment created for 'main'
segdef created for segment main, symbol 'main', addr:000000
segref created for segment 'UtilsA' symbol:ioa, addr:000304
segref created for segment 'UtilsA' symbol:rd_clockA, addr:000306
segref created for segment 'lot_' symbol:lot_, addr:000310
segref created for segment 'stack_0' symbol:stack_0, addr:000312
segref created for segment 'power_of_two' symbol:power_of_two, addr:000314
segref created for segment 'power_of_two' symbol:power_of_two2, addr:000316
segref created for segment 'simh_operators_' symbol:operator_table, addr:000320
segref created for segment 'simh_operators_' symbol:operator_table+632, addr:000322
segref created for segment 'simh_operators_' symbol:operator_table+633, addr:000324
segref created for segment 'simh_operators_' symbol:operator_table+634, addr:000326
segref created for segment 'simh_operators_' symbol:operator_table+635, addr:000330
segref created for segment 'simh_operators_' symbol:operator_table+636, addr:000332
Writing SDW to address 00001000 (DSBR.ADDR+2*0 offset) 
213 (000325) words loaded into segment 0(0) at address 000000
segment created for 'Play'
segdef created for segment Play, symbol 'Main', addr:000003
segref created for segment 'UtilsA' symbol:A0, addr:000226
segref created for segment 'UtilsA' symbol:A1, addr:000230
segref created for segment 'UtilsA' symbol:rd_clockA, addr:000232
segref created for segment 'UtilsA' symbol:ioa, addr:000234
154 (000232) words loaded into segment Play
segment created for 'UtilsA'
segdef created for segment UtilsA, symbol 'A0', addr:000005
segdef created for segment UtilsA, symbol 'A1', addr:000050
segdef created for segment UtilsA, symbol 'rd_clockA', addr:000207
segdef created for segment UtilsA, symbol 'ioa', addr:000270
851 (001523) words loaded into segment UtilsA
segment created for 'simh_operators_'
segdef created for segment simh_operators_, symbol 'alm_call', addr:001175
segdef created for segment simh_operators_, symbol 'alm_entry', addr:001215
segdef created for segment simh_operators_, symbol 'alm_operators_begin', addr:001175
segdef created for segment simh_operators_, symbol 'alm_operators_end', addr:001237
segdef created for segment simh_operators_, symbol 'alm_push', addr:001201
segdef created for segment simh_operators_, symbol 'alm_return', addr:001225
segdef created for segment simh_operators_, symbol 'alm_return_no_pop', addr:001233
segdef created for segment simh_operators_, symbol 'transfer_vector', addr:000551
segdef created for segment simh_operators_, symbol 'operator_table', addr:000000
40 (000050) words loaded into segment simh_operators_
segment created for 'power_of_two'
entrypoint created for segment main, symbol 'power_of_two', addr:000032
entrypoint created for segment main, symbol 'power_of_two2', addr:000034
30 (000036) words loaded into segment power_of_two
lot_ segment created with 5 sparse entries.
stack_0 segment created as segment# 8 (10) [DSBR.STACK=0001]
Examining segments ... segment numbering begins at: 9
    Processing segment main...
        Resolving segref UtilsA$ioa...found ioa (000304)
            ITS Pair: [even:000011000043, odd:000270000000]
        Resolving segref UtilsA$rd_clockA...found rd_clockA (000306)
            ITS Pair: [even:000011000043, odd:000207000000]
        Resolving segref lot_$lot_...found lot_ (000310)
            ITS Pair: [even:000012000043, odd:000000000000]
        Resolving segref stack_0$stack_0...found stack_0 (000312)
            ITS Pair: [even:000010000043, odd:000000000000]
        Resolving segref power_of_two$power_of_two...found power_of_two (000314)
            ITS Pair: [even:000013000043, odd:000032000000]
        Resolving segref power_of_two$power_of_two2...found power_of_two2 (000316)
            ITS Pair: [even:000013000043, odd:000034000000]
        Resolving segref simh_operators_$operator_table...found operator_table (000320)
            ITS Pair: [even:000014000043, odd:000000000000]
        Resolving segref simh_operators_$operator_table...found operator_table+632 (000322)
            ITS Pair: [even:000014000043, odd:001170000000]
        Resolving segref simh_operators_$operator_table...found operator_table+633 (000324)
            ITS Pair: [even:000014000043, odd:001171000000]
        Resolving segref simh_operators_$operator_table...found operator_table+634 (000326)
            ITS Pair: [even:000014000043, odd:001172000000]
        Resolving segref simh_operators_$operator_table...found operator_table+635 (000330)
            ITS Pair: [even:000014000043, odd:001173000000]
        Resolving segref simh_operators_$operator_table...found operator_table+636 (000332)
            ITS Pair: [even:000014000043, odd:001174000000]
    Processing segment Play...
        Resolving segref UtilsA$A0...found A0 (000226)
            ITS Pair: [even:000011000043, odd:000005000000]
        Resolving segref UtilsA$A1...found A1 (000230)
            ITS Pair: [even:000011000043, odd:000050000000]
        Resolving segref UtilsA$rd_clockA...found rd_clockA (000232)
            ITS Pair: [even:000011000043, odd:000207000000]
        Resolving segref UtilsA$ioa...found ioa (000234)
            ITS Pair: [even:000011000043, odd:000270000000]
    Processing segment UtilsA...
    Processing segment simh_operators_...
    Processing segment power_of_two...
    Processing segment lot_...
    Processing segment stack_0...
Loading deferred segments ...
    loading Play as segment# 13
Writing SDW to address 00001032 (DSBR.ADDR+2*13 offset) 
      158 (000236) words loaded into segment 13 (15) at address 201000
    loading UtilsA as segment# 9
Writing SDW to address 00001022 (DSBR.ADDR+2*9 offset) 
      1058 (002042) words loaded into segment 9 (11) at address 201240
    loading simh_operators_ as segment# 12
Writing SDW to address 00001030 (DSBR.ADDR+2*12 offset) 
      671 (001237) words loaded into segment 12 (14) at address 203320
    loading power_of_two as segment# 11
Writing SDW to address 00001026 (DSBR.ADDR+2*11 offset) 
      30 (000036) words loaded into segment 11 (13) at address 204560
    loading lot_ as segment# 10
Writing SDW to address 00001024 (DSBR.ADDR+2*10 offset) 
      4095 (007777) words loaded into segment 10 (12) at address 204620
LOT => PR[4]: SNR=00012 RNR=0 WORDNO=000000 BITNO:00
    loading stack_0 as segment# 8
Writing SDW to address 00001020 (DSBR.ADDR+2*8 offset) 
      49152 (140000) words loaded into segment 8 (10) at address 214620
snapping lot_ links..
Howdy world! - It's '10/29 20:35 '
We did it!!! 2**9=512
We did it again!!! 2**11=2048
Q=1050402 (Should be Q=1050402)
Q=60606 (Should be Q=60606)
Q=10703020107 (Should be Q=010703020107)

cpuCycles = 4300
TestFXE-27> g 0

DIS instruction, IC: main|162 (00000162) (000000616000)
Goodbye
